### PR TITLE
OLD: assert get.---()

### PR DESCRIPTION
**Pull request by grewered**
line 2512: added assertions for get.plausible()
I didn't touch the other lines of code, but R marked them as edited.

### CODE REVIEW

File: R/eatModel.r [hunk 1]
```diff
@@ -2509,34 +2509,38 @@ wleFromRes <- function ( resultsObj , idVarName = NULL, verbose=TRUE) {
              return(sel)
           }  }
 
-get.plausible <- function(file, quiet = FALSE, forConquestResults = FALSE)  {   
+get.plausible <- function(file, quiet = FALSE, forConquestResults = FALSE)  {
+  # checks
+  checkmate::assert_character(file)
```
Comment (weirichs) [inline R/eatModel.r]: I think you coud add `checkmate::assert_character(file, len=1)` here.

File: R/eatModel.r [hunk 2]
```diff
@@ -2509,34 +2509,38 @@ wleFromRes <- function ( resultsObj , idVarName = NULL, verbose=TRUE) {
              return(sel)
           }  }
 
-get.plausible <- function(file, quiet = FALSE, forConquestResults = FALSE)  {   
+get.plausible <- function(file, quiet = FALSE, forConquestResults = FALSE)  {
+  # checks
+  checkmate::assert_character(file)
+  lapply(c(quiet, forConquestResults), checkmate::assert_logical, len = 1)
```
Comment (weirichs) [inline R/eatModel.r:2515]: thanks, LGTM

**Review by weirichs — COMMENTED** [review]
I've added one little comment directly in the text.

File: R/eatModel.r [hunk 3]
```diff
@@ -2552,60 +2555,67 @@ get.plausible <- function(file, quiet = FALSE, forConquestResults = FALSE)  {
...
             colnames(valid)[(ncol(valid)-length(namen2)):1] <- c("ID","case")[1:(ncol(valid)-length(namen2))]
             colnames(valid)[(ncol(valid)-length(namen2)+1):ncol(valid)] <- paste(namen1,namen2,sep="")
             return(valid)}
 
 get.shw <- function(file, dif.term, split.dif = TRUE, abs.dif.bound = 0.6, sig.dif.bound = 0.3, p.value = 0.9) {
-            all.output <- list();   all.terms <- NULL                           
-            input.all <- scan(file,what="character",sep="\n",quiet=TRUE)        
+  # checks
+  checkmate::assert_character(file, len = 1)
+  checkmate::assert_character(dif.term, null.ok = TRUE)
```
Comment (grewered) [inline R/eatModel.r]: shall I check here, whether string in "file" contains "dif.term"?

File: R/eatModel.r
Comment (weirichs) [inline R/eatModel.r:1]: In it's current form, users may omit the `dif.term` argument. With assertions implemented, however, an error message like 
```
Fehler in checkmate::assert_character(dif.term, null.ok = TRUE) : 
  argument "dif.term" is missing, with no default
```
probably will occur. I would prefer to add the assertion `checkmate::assert_character(dif.term, len = 1)` at a later point in the function, i.e. after the occurrence `if(!missing(dif.term))`

**Review by weirichs — COMMENTED** [review]
`checkmate::assert_character(dif.term)` should be modified, see above.

**Comment by grewered** [comment]
ok, I changed dif.term. the other get. functions only have a single argument, I wrote the same check for every one. 
can I merge? 
Or do you know, why the checks fail? but I think that was like that before I started.

**Review by weirichs — APPROVED** [review]
Thanks, LGTM!

**Comment by weirichs** [comment]
> ok, I changed dif.term. the other get. functions only have a single argument, I wrote the same check for every one. can I merge? Or do you know, why the checks fail? but I think that was like that before I started.

Actually I don't know why the checks fail. But you're right, that already happened before. Feel free to merge! Thanks a lot!

File: R/eatModel.r (hunk 3 above)
Comment (weirichs) [reply R/eatModel.r]: The `get.shw` function is not very well written, to be honest. See, for example the statement `ind <- grep("TERM",input.all)` which looks for all occurrences of `TERM` within the ASCII file. The result of `ind` must be a numerical vector with at least one element:
`if(length(ind)==0) {stop(paste("No TERM-statement found in file ",file,".\n",sep=""))}` could be replaced with `assert_numeric(ind, min.len = 1, unique=TRUE)`. However, if these check fails, a user message like `Assertion on 'ind' failed: Must have length >= 1, but has length 0.` is not very informative for the user, but I don't know whether checkmate allows to customize message wording. 

If `dif.term` is not missing, `dif.term` should occur in `input.all`. Maybe a statement like 
```
!missing(dif.term){
  ind2 <- grep(dif.term, input.all, fixed = TRUE)
  assert_numeric(ind2, len = 1) }
```
could be useful?

`fixed = TRUE` is necessary here as `dif.term` contains the conquest statement for differential item functioning (DIF) which ofen occurs special characters (asterisks). Without `fixed = TRUE`, statements like `item*gender` will not be found.

File: R/eatModel.r [hunk 4]
```diff
@@ -2635,15 +2644,16 @@ get.shw <- function(file, dif.term, split.dif = TRUE, abs.dif.bound = 0.6, sig.d
...
-                    colnames(input.sel) <- namen                                
+                    colnames(input.sel) <- namen
                     input.sel  <- suppressWarnings(eatTools::asNumericIfPossible(data.frame( gsub("NNA", NA, gsub("NA", NA, gsub("\\*","",input.sel))), stringsAsFactors = FALSE), force.string = FALSE))
                     results.sel<- data.frame(input.sel,filename=file,stringsAsFactors = FALSE)
                     if(is.na(as.numeric(results.sel$ESTIMATE[1]))) {cat(paste("'ESTIMATE' column in Outputfile for term '",all.terms[length(all.terms)],"' in file: '",file,"' does not seem to be a numeric value. Please check!\n",sep=""))}
-                    if(!missing(dif.term)) {                                    
-                       if(all.terms[length(all.terms)] == dif.term) {           
+                    if(!missing(dif.term)) {
+                      checkmate::assert_character(dif.term, len = 1)
+                       if(all.terms[length(all.terms)] == dif.term) {
```
Comment (grewered) [inline R/eatModel.r:2654]: l. 2653:
so this is fine or should I add/change what you suggested here?


```
The get.shw function is not very well written, to be honest. See, for example the statement ind <- grep("TERM",input.all) which looks for all occurrences of TERM within the ASCII file. The result of ind must be a numerical vector with at least one element:
if(length(ind)==0) {stop(paste("No TERM-statement found in file ",file,".\n",sep=""))} could be replaced with assert_numeric(ind, min.len = 1, unique=TRUE). However, if these check fails, a user message like Assertion on 'ind' failed: Must have length >= 1, but has length 0. is not very informative for the user, but I don't know whether checkmate allows to customize message wording.

If dif.term is not missing, dif.term should occur in input.all. Maybe a statement like

!missing(dif.term){
  ind2 <- grep(dif.term, input.all, fixed = TRUE)
  assert_numeric(ind2, len = 1) }

could be useful?

fixed = TRUE is necessary here as dif.term contains the conquest statement for differential item functioning (DIF) which ofen occurs special characters (asterisks). Without fixed = TRUE, statements like item*gender will not be found. 
```

Comment (grewered) [reply R/eatModel.r:2654]: @weirichs

Comment (weirichs) [reply R/eatModel.r:2654]: Your additions are good, but I would suggest adding the following between the lines 2652 and 2653: 
```
ind2 <- grep(dif.term, input.all, fixed = TRUE)
assert_numeric(ind2, len = 1)
```
Thanks a lot!

**Review by weirichs — APPROVED** [review]
These changes are too extensive to be checked by hand. I would suggest using `devtools::check()` and `devtools::test()` here.

**Pull request closed without merging**